rust/operator-binary/src/config.rs

```diff
@@ -0,0 +1,184 @@
+
+    config.push_str(
+        formatdoc! {r#"
+                # Authentication configs
+                AUTH_TYPE = AUTH_LDAP
+                AUTH_LDAP_SERVER = "{protocol}{server_hostname}:{server_port}"
+
+                AUTH_LDAP_SEARCH = "{search}"
+                AUTH_LDAP_SEARCH_FILTER = "{search_filter}"
+                AUTH_LDAP_UID_FIELD = "{uid_field}"
+                AUTH_LDAP_GROUP_FIELD = "{group_field}"
+                AUTH_LDAP_FIRSTNAME_FIELD = "{firstname_field}"
+                AUTH_LDAP_LASTNAME_FIELD = "{lastname_field}"
+                AUTH_LDAP_EMAIL_FIELD = "{email_field}"
+
+                AUTH_USER_REGISTRATION = {user_registration}
+                AUTH_USER_REGISTRATION_ROLE = "{user_registration_role}"
+                AUTH_ROLES_SYNC_AT_LOGIN = {roles_sync_at_login}
+            "#,
+            protocol = match ldap.tls {
+                None => "ldap://",
+                Some(_) => "ldaps://",
+            },
+            server_hostname = ldap.hostname,
+            server_port = ldap.port.unwrap_or_else(|| ldap.default_port()),
+            search = ldap.search_base,
+            search_filter = ldap.search_filter,
+            uid_field = ldap.ldap_field_names.uid,
+            group_field = ldap.ldap_field_names.group,
+            firstname_field = ldap.ldap_field_names.given_name,
+            lastname_field = ldap.ldap_field_names.surname,
+            email_field = ldap.ldap_field_names.email,
+            user_registration = to_python_bool(
+                authentication_method
+                    .ldap_extras
+                    .as_ref()
+                    .map(|extra| extra.user_registration)
+                    .unwrap_or_else(stackable_superset_crd::default_user_registration)
+            ),
+            user_registration_role = authentication_method
+                .ldap_extras
+                .as_ref()
+                .map(|extra| &extra.user_registration_role)
+                .unwrap_or(&stackable_superset_crd::default_user_registration_role()),
+            roles_sync_at_login = to_python_bool(
+                authentication_method
+                    .ldap_extras
+                    .as_ref()
+                    .map(|extra| &extra.sync_roles_at)
+                    .unwrap_or(&stackable_superset_crd::default_sync_roles_at())
+                == &LdapRolesSyncMoment::Login
+            ),
+        }
+        .as_str(),
+    );
+
+    // Possible TLS options, see https://github.com/dpgaspar/Flask-AppBuilder/blob/f6f66fc1bcc0163a213e4a2e6f960e91082d201f/flask_appbuilder/security/manager.py#L243-L250
+    match &ldap.tls {
+        None => config.push_str("AUTH_LDAP_USE_TLS = False\n"),
+        Some(tls) => match &tls.verification {
+            TlsVerification::None {} => {
+                config.push_str(indoc! {r#"
+                    AUTH_LDAP_USE_TLS = False # This setting is for STARTTLS which we dont support at the moment, we use dedicated Tls.
```

Comment:
Can we move this to be outside of the `match &tls.verification` block, since this is the same for all variants?

Reply:
I actually remove the STARTTLS setting completely